File: cppe/demo/IceE/callback/CallbackSenderI.cpp
// **********************************************************************
//
// Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
//
// This copy of Ice-E is licensed to you under the terms described in the
// ICEE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <CallbackSenderI.h>
#include <IceE/IceE.h>

using namespace std;
using namespace Demo;

void
CallbackSenderI::initiateCallback(const CallbackReceiverPrx& proxy, const Ice::Current& current)
{
    printf("initiating callback\n");
    try
    {
	proxy->callback(current.ctx);
    }
    catch(const Ice::Exception& ex)
    {
	fprintf(stderr, "%s\n", ex.toString().c_str());
    }
}

void
CallbackSenderI::shutdown(const Ice::Current& c)
{
    printf("shutting down...\n");
    try
    {
	c.adapter->getCommunicator()->shutdown();
    }
    catch(const Ice::Exception& ex)
    {
	fprintf(stderr, "%s\n", ex.toString().c_str());
    }
}

File: javae/demo/IceE/jdk/callback/CallbackSenderI.java
// **********************************************************************
//
// Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
//
// This copy of Ice-E is licensed to you under the terms described in the
// ICEE_LICENSE file included in this distribution.
//
// **********************************************************************

import Demo.*;

public final class CallbackSenderI extends _CallbackSenderDisp
{
    public void
    initiateCallback(CallbackReceiverPrx proxy, Ice.Current current)
    {
        System.out.println("initiating callback");
	try
	{
	    proxy.callback(current.ctx);
	}
	catch(Ice.LocalException ex)
	{
	    ex.printStackTrace();
	}
    }

    public void
    shutdown(Ice.Current current)
    {
        System.out.println("Shutting down...");
	try
	{
	    current.adapter.getCommunicator().shutdown();
	}
	catch(Ice.LocalException ex)
	{
	    ex.printStackTrace();
	}
    }
}
